### PR TITLE
Link the author to the author show page

### PR DESCRIPTION
**Pull request by jcoyne**
when viewing publications, authors are connected to publications and are viewable/linked (from publications --> authors)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ gem 'pg'
 # Use Puma as the app server
 gem 'puma', '~> 3.11'
 
-gem 'blacklight', github: 'projectblacklight/blacklight'
+gem 'blacklight', github: 'projectblacklight/blacklight', branch: 'avoid_unnecessary_lookup'
```
Comment (justinlittman) [inline Gemfile]: TODO: Add ticket for undoing this.

File: Gemfile.lock [hunk 2]
```diff
@@ -1,6 +1,7 @@
 GIT
   remote: https://github.com/projectblacklight/blacklight.git
-  revision: 94749d5aa34c4474f4ad27c1dc35f95442271f23
+  revision: d3e9eb1977ca81e6d5737870524a3466dd29ad93
+  branch: avoid_unnecessary_lookup
```
Comment (justinlittman) [inline Gemfile.lock]: And this ...

File: app/models/solr_document.rb [hunk 3]
```diff
@@ -17,4 +17,19 @@ class SolrDocument
   # and Blacklight::Document::SemanticFields#to_semantic_values
   # Recommendation: Use field names from Dublin Core
   use_extension(Blacklight::Document::DublinCore)
+
+  # Produces a list of authors with links to the frontend app for each.
+  def linked_authors
+    foo = self['author_labels_tsim'].zip(self['authors_ssim'])
```
Comment (justinlittman) [inline app/models/solr_document.rb]: authors or something else instead of foo?

File: Gemfile.lock (hunk 2 above)
Comment (justinlittman) [inline Gemfile.lock]:
```
li-dl-7328-02d5:rialto-webapp jlittman$ bundle install
Fetching gem metadata from https://rubygems.org/.......
Fetching https://github.com/projectblacklight/blacklight.git
fatal: Could not parse object 'd3e9eb1977ca81e6d5737870524a3466dd29ad93'.
Revision d3e9eb1977ca81e6d5737870524a3466dd29ad93 does not exist in the repository https://github.com/projectblacklight/blacklight.git. Maybe you
misspelled it?
```

File: Gemfile (hunk 1 above)
Comment (jcoyne) [reply Gemfile]: https://github.com/sul-dlss-labs/rialto-webapp/issues/93

File: Gemfile.lock (hunk 2 above)
Comment (jcoyne) [reply Gemfile.lock]: I resolved this with a bundle update. That branch got rebased.

**Comment by justinlittman** [comment]
For retrieving a publication (http://localhost:8080/#/catalog/http:%2F%2Fsul.stanford.edu%2Frialto%2Fpublications%2Fb64e617dc2e10dc60826b4aeb1ff0c9b), I'm getting:
```
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d] Solr fetch (26.6ms)
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d] method=GET path=/catalog/http%3A%2F%2Fsul.stanford.edu%2Frialto%2Fpublications%2Fb64e617dc2e10dc60826b4aeb1ff0c9b format=json controller=CatalogController action=show status=500 error='ActionView::Template::Error: undefined method `zip' for nil:NilClass' duration=166.22 view=0.00 db=0.00
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d]   
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d] ActionView::Template::Error (undefined method `zip' for nil:NilClass):
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d]     4:   json.id "#{document_url}##{field_name}"
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d]     5:   json.type 'document_value'
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d]     6:   json.attributes do
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d]     7:     json.value doc_presenter.field_value(field)
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d]     8:     json.label field.label
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d]     9:   end
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d]    10: end
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d]   
webapp_1                  | [b58e25d9-db04-4cf2-a9eb-74157e76719d] app/models/solr_document.rb:23:in `linked_authors'
```

File: app/models/solr_document.rb [hunk 4]
```diff
@@ -17,4 +17,19 @@ class SolrDocument
   # and Blacklight::Document::SemanticFields#to_semantic_values
   # Recommendation: Use field names from Dublin Core
   use_extension(Blacklight::Document::DublinCore)
+
+  # Produces a list of authors with links to the frontend app for each.
+  def linked_authors
+    tuples = self['author_labels_tsim'].zip(self['authors_ssim'])
+    parts = tuples.map do |author|
+      "<a href=\"#{search_link(author.last)}\">#{author.first}</a>".html_safe
```
Comment (mjgiarlo) [inline app/models/solr_document.rb]: Confirming: you need this `.html_safe` *and* the one on line 27? I'd have naïvely assumed the one on line 27 would take care of it, once all the strings in the list of tuples are joined into a single string.

Comment (jcoyne) [reply app/models/solr_document.rb]: `Array#to_sentence` knows nothing about html safety, so we need the one on line 27, however, we probably don't need the one on line 25.

**Pull request merged**